src/components/InputField.vue

```diff
@@ -0,0 +1,67 @@
+<script setup>
+import { ref, onMounted } from 'vue'
+
+const { name } = defineProps({
+  name: String,
+  error: String
+})
+
+const model = defineModel()
+
+const inputLabel = ref('')
+const inputType = ref('')
+const inputPlaceholder = ref('')
+const inputIcon = ref('')
+
+onMounted(() => {
+  switch (name) {
+    case 'name':
+      inputLabel.value = 'Author Name'
+      inputType.value = 'text'
+      inputPlaceholder.value = 'Name Surname'
+      inputIcon.value = 'fa-user'
+      break
+    case 'email':
+      inputLabel.value = 'Author Email'
+      inputType.value = 'email'
+      inputPlaceholder.value = 'Your Email'
+      inputIcon.value = 'fa-envelope'
+      break
+
+    case 'title':
+      inputLabel.value = 'Title'
+      inputType.value = 'text'
+      inputPlaceholder.value = 'Post Title'
+      inputIcon.value = 'fa-user'
+      break
+
+    default:
+      break
+  }
+})
```

Comment:
You have multiple refs like `inputLabel`, `inputType`, `inputPlaceholder`, and `inputIcon`, but they are only updated based on the name prop. Using computed properties is more efficient here since the values are derived from a reactive source (name) and don't need to be set on `onMounted().`